### DISCUSSION TITLE
Temporary turn off Kafka brokers

### DISCUSSION THREAD

**Discussion by Delorien84** [strimzi/strimzi-kafka-operator#6989] [Q&A] [answered]
Hello,
 I would like to ask, if there is a way how to temporary turn of all Kafka brokers. My use case is to shut down whole testing environment when it is not needed (saving money). Currently I just delete whole `Kafka` resource with setup of keeping PVC. When then I recreate the `Kafka` resource the brokers turn on with all the data. Problem is that, destroying and recreating `Kafka` resource will re-create whole certificate chain. It is not a problem with application inside Kubernetes cluster, but we also use application outside kubernetes cluster. It requires to export Kafka certificates outside Kubernetes and that is manual work that we want to avoid.

Is there some way to temporary turn off Kafka brokers but keep certificate chain intact?

Thank you

**Comment by scholzj** [comment, marked as answer]
I'm not sure there is a straight forward solution for this. I guess there are several things you can try:
* Disabling the owner reference for the CA certificates which can be done as part of the Kafka CR. Then they should not be deleted when you delete the Kafka CR. (Or using Custom CAs might be another alternative to this)
* Pausing the reconciliation of the Kafka CR (there is an annotation which can do it) and scaling down or deleting the Deployments and StatefulSets or deleting the StrimziPodSets (which cannot be scaled down) to just get rig of the Pods. When you unpause it, Strimzi should just recreate them, but other resources (Secrets, services, etc.) should keep exist as before.

**Reply by rareddy** [reply]
we have a similar need in RHOSAK, what is the feasibility of this offering this as a feature of Strimzi? I am not looking to delete any PVs, secrets, certs, or config maps. Just scale down the compute resources and resume/resurrect later.

**Reply by scholzj** [reply]
I'm not aware of any plans for anything more sophisticated than this. What is your use-case?

**Reply by rareddy** [reply]
The use-case is the ability to freeze an instance in extreme situations when payments are overdue and we want to reduce our capital costs.

**Reply by Delorien84** [reply]
@scholzj Thank for idea about stop the reconciliation of the Kafka CR. It is enough for us. I will implement it. Thank you.

**Reply by tombentley** [reply]
It might not be a problem for this testing use case, but for related hibernation use cases the deletion of topic data due to Kafka topic retention configs should be considered: E.g. you hibernate for a week and end up with no data in your cluster.
